api/oneclick.py

```diff
@@ -211,105 +522,68 @@ def get_patron_information(self, patron_id):
-
-        patron = PatronInfo(username=resp_dict['userName'], card_number=resp_dict['libraryCardNumber'],  
-            email=resp_dict['email'], first_name=resp_dict['firstName'], last_name=resp_dict['lastName'])
-
-        return patron
-
+        self.validate_response(response, message, action=action)
 
-    def fulfill(self, patron, pin, licensepool, internal_format):
-        """ Get the actual resource file to the patron.
-        :return a FulfillmentInfo object.
-        """
-        raise NotImplementedException
+        # If needed, will put info into PatronData subclass.  For now, OK to return a dictionary.
+        return resp_dict
 
 
     def patron_activity(self, patron, pin):
-        """ Return a patron's current checkouts and holds.
+        """ Get a patron's current checkouts and holds.
 
         :param patron: a Patron object for the patron who wants to return the book.
         :param pin: The patron's password (not used).
         """
-        (patron_oneclick_id, item_oneclick_id) = self.validate_input(patron, licensepool=None, patron_only=True)
+        patron_oneclick_id = self.validate_patron(patron)
 
-        url = "%s/libraries/%s/patrons/%s/checkouts/" % (self.base_url, str(self.library_id), patron_id)
-        action="patron_activity"
-        
-        try:
-            response = self.request(url=url)
-        except Exception, e:
-            self.log.error("Patron info failed: %r", e, exc_info=e)
-            raise RemoteInitiatedServerError(e.message, action)
-
-        resp_dict = {}
-        message = None
-        if response.text:
-            resp_dict = response.json()
-            message = resp_dict.get('message', None)
+        patron_checkouts = self.get_patron_checkouts(patron_oneclick_id)
+        patron_holds = self.get_patron_holds(patron_oneclick_id)
 
-        error_response = self.validate_response(response=response, message=message, action=action)
+        return (patron_checkouts, patron_holds)
 
-        # TODO: go through patron's checkouts and holds and 
-        # generate LoanInfo and HoldInfo objects.
-        loan = LoanInfo(
-            identifier.type,
-            item_oneclick_id,
-            today,
-            expires,
-            None,
-        )
 
-        hold = HoldInfo(
-            Identifier.OVERDRIVE_ID,
-            overdrive_identifier,
-            start_date=start,
-            end_date=end,
-            hold_position=position
-        )
+    ''' -------------------------- Validation Handling -------------------------- '''
+    def validate_item(self, licensepool):
+        """ Are we performing operations on a book that exists and can be 
+        uniquely identified? 
+        """
+        item_oneclick_id = None
+        media = None
 
+        identifier = licensepool.identifier
+        item_oneclick_id=identifier.identifier
+        if not item_oneclick_id:
+            raise InvalidInputException("Licensepool %r doesn't know its ISBN.", licensepool)
```

Comment:
This seems like a much lower-level test than anything you should need to worry about. Why would you end up with a Licensepool whose Identifier had a blank .identifier? And wouldn't its identifier be an internal OneClick ID, not an ISBN?